tests/tools/test_samtools.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import pathlib
+from collections import OrderedDict
+
+import pysam
+import fastq
+
+
+@pytest.mark.workflow('samtools_split')
+def test_samtools_split(workflow_dir):
+    bam = pathlib.Path(workflow_dir, 'test-output/out/split_bams/0/test.1.bam')
+    samfile = pysam.AlignmentFile(bam, "rb")
+    bam_header = OrderedDict((k, v) for k, v in samfile.header.items())
+    read_groups = [read_group['ID'] for read_group in bam_header.get('RG', []) if 'ID' in read_group]
+    assert len(read_groups) == 1
+    assert read_groups[0] == "1"
+
+    second_bam = pathlib.Path(workflow_dir, 'test-output/out/split_bams/1/test.2.bam')
+    second_samfile = pysam.AlignmentFile(second_bam, "rb")
+    second_bam_header = OrderedDict((k, v) for k, v in second_samfile.header.items())
+    second_read_groups = [read_group['ID'] for read_group in second_bam_header.get('RG', []) if 'ID' in read_group]
+    assert len(second_read_groups) == 1
+    assert second_read_groups[0] == "2"
+
+@pytest.mark.workflow('samtools_merge')
+def test_samtools_merge(workflow_dir):
+    bam = pathlib.Path(workflow_dir, 'test-output/out/merged_bam/test.bam')
+    samfile = pysam.AlignmentFile(bam, "rb")
+    bam_header = OrderedDict((k, v) for k, v in samfile.header.items())
+    read_groups = [read_group['ID'] for read_group in bam_header.get('RG', []) if 'ID' in read_group]
+    assert len(read_groups) == 2
+    assert read_groups[0] == "test2"
+    assert read_groups[1] == "test.bwa_aln_pe"
+
+@pytest.mark.workflow('samtools_collate', 'samtools_collate_to_fastq')
+def test_samtools_collate(workflow_dir):
+    bam = pathlib.Path(workflow_dir, 'test-output/out/collated_bam/test.bwa_aln_pe.collated.bam')
+    samfile = pysam.AlignmentFile(bam, "rb")
+
+    reads = list(samfile.head(n=100))
```

Comment:
I think this BAM is small enough that we could read the entire thing. Is that worth a little runtime penalty? If some rare case caused it to be not correctly `collated`, we probably wouldn't catch it anyway with a BAM this size.

Reply:
We might as well read the whole thing. I doubt it will be noticeable given the ~15min this already takes.